getting-started/03.ACTIONS.md

```diff
@@ -0,0 +1,64 @@
+# Actions
+
+1. [Prompts](./00.PROMPTS.md)
+2. [Prompt Template](./01.TEMPLATES.md)
+3. [Planner](./02.PLANNER.md)
+4. [**Actions**](./03.ACTIONS.md)
+5. [Chain](./04.CHAIN.md)
+6. [Turns](./05.TURNS.md)
+
+An action is an atomic function that is registered to the AI Module. It is a fundamental building block of a plan. 
+
+Here's an example of how the `createList` action would look like in code:
+
+### C#
+
+```dotnet
+[Action("createList")]
+public bool CreateList([ActionTurnState] ListState turnState, [ActionEntities] Dictionary<string, object> entities)
+{
+    ArgumentNullException.ThrowIfNull(turnState);
+    ArgumentNullException.ThrowIfNull(entities);
+
+    string listName = GetEntityString(entities, "list");
+
+    // Ex. create a list with name "Grocery Shopping"
+    CreateList(listName);
+
+    // Continues exectuion of next command in the plan.
+    return true;
+}
+```
+
+>Adding the `Action` attribute marks the method as an action. To register it to the AI module you have pass the instance object containing this method to the `AI.ImportActions(instance)` method. Alternatively you can use the `AI.RegisterAction(name, handler)` to register a single action.   
+
+### JS/TS
+
+```JS
+app.ai.action('createList', async (context: TurnContext, state: ApplicationTurnState, data: EntityData) => {
+    // Ex. create a list with name "Grocery Shopping".
+    createList(data.name);
+
+    // Continues exectuion of next command in the plan. 
+    return true;
+});
+```
+>The `action` method registers the action named `createList` with corresponding callback function.
+
+## Default Actions
+
+The user can register custom actions or override default actions in the system. Below is a list of default actions present:
+
+| Action   | Description  |
+|---|---|
+| `__UnkownAction___`  | Called anytime an unknown action is predicted by the planner.   |
+| `___FlaggedInput___`  | Called anytime an input is flagged by the moderator.   | 
+| `___FlaggedOutput___`  | Called anytime an output is flagged by the moderator.   | 
+| `___RateLimited___`  | Called anytime the planner is rate limited.   | 
+| `___PlanReady___`  | Called after the plan has been predicted by the planner and it has passed moderation.   | 
+| `___DO___`  | Called to DO an action.   | 
```

Comment:
I would leave a note that changing these will change ALL `DO` and `SAY` actions (as in, it's global as opposed to one-time).

Reply:
done